src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -300,6 +298,14 @@ public static KNNQueryBuilder fromXContent(XContentParser parser) throws IOExcep
 
         validateSingleQueryType(k, maxDistance, minScore);
 
+        updateQueryStats(k, filter, List.of(KNNCounter.KNN_QUERY_REQUESTS, KNNCounter.KNN_QUERY_WITH_FILTER_REQUESTS));
+        updateQueryStats(minScore, filter, List.of(KNNCounter.MIN_SCORE_QUERY_REQUESTS, KNNCounter.MIN_SCORE_QUERY_WITH_FILTER_REQUESTS));
+        updateQueryStats(
+            maxDistance,
+            filter,
+            List.of(KNNCounter.MAX_DISTANCE_QUERY_REQUESTS, KNNCounter.MAX_DISTANCE_QUERY_WITH_FILTER_REQUESTS)
+        );
```

Comment:
rather than these 3 functions you can create a list of all the counters needs to be updated during the parsing of the Query. This is what I was saying and then in 1 single shot you can basically set all the counters.

Reply:
"a list of all the counters" might not applicable.. because the counter usage logic of QUERY_REQUESTS and QUERY_WITH_FILTER_REQUESTS are different.

Reply:
Had offline talk with @navneet1v, introduced VectorQueryType enum to handle which counter to use.